### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by Ademola101**
This project contains various enumerable methods in the Enumerable module. The methods were all implemented by me with aid of ruby techniques that I have learnt so far

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+  def my_count(arg = nil)
+    sum = 0
+    if block_given?
+      my_each { |a| sum += 1 if yield(a) }
+    elsif arg.nil?
+      my_each { sum += 1 }
+    else
+      my_each { sum += 1 if a == arg }
+    end
```
Comment (soieducation) [inline enumerable.rb]: - [ ] The return statement here is missing the "a" variable in the my_each block

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    while i < collection.length
+      block_return_values << yield(collection[i])
+      i += 1
+    end
+    if block_return_values.include?(false)
+      false
+    else
+      true
+    end
+  end
```
Comment (soieducation) [inline enumerable.rb]: - [ ] This method doesn't return anything when its called with an array, kindly fix it.

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  def my_any?
+    my_each do |item|
+      if block_given?
+        return true if yield item
+      elsif item
+        return true
+      end
+    end
+    false
+  end
```
Comment (soieducation) [inline enumerable.rb]: - [ ] my_any? method should test for Regexp too. Check test cases [here](https://ruby-doc.org/core-2.6.1/Enumerable.html#method-i-any-3F)
- [ ] my_any? method should test for Classes too. Check test cases [here](https://ruby-doc.org/core-2.6.1/Enumerable.html#method-i-any-3F)

File: enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    end
+    sum
+  end
+
+  def my_none?
+    my_each do |item|
+      return false if block_given? && yield(item) || !block_given? && item
+    end
+    true
+  end
```
Comment (soieducation) [inline enumerable.rb]: - [ ] my_none? method should test for Regexp too. Check test cases [here](https://ruby-doc.org/core-2.6.1/Enumerable.html#method-i-none-3F)
- [ ] my_none? method should test for Classes too. Check test cases [here](https://ruby-doc.org/core-2.6.1/Enumerable.html#method-i-none-3F)

File: enumerable.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+  def my_inject(accumulator, &block)
+    each do |element|
+      accumulator = block.call(accumulator, element)
+    end
+    accumulator
+  end
```
Comment (soieducation) [inline enumerable.rb]: - [ ] Refactor my_inject to pass all tests cases in the documentation given above.

**Review by soieducation — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🚦 

## Hello @Ademola101  :handshake: :handshake: :handshake: 
 Great work on your first attempt
### However, there are still changes to be made

# [Required Changes] :warning: 
- Check inline comments below
___
**If you are stuck at any point or have any questions, let us know by leaving a comment here. We will be happy to help**
HAPPY CODING :computer:

**Review by soieducation — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🚦 

## Hello @Ademola101  :handshake: :handshake: :handshake: 
 Great work so far

### However, there are still changes to be made

# [Required Changes] :warning: 
- [ ] All methods are not passing except `my_each`, `my_each_with_index` and `my_select`. Kindly refactor and test again.
___
**If you are stuck at any point or have any questions, let us know by leaving a comment here. We will be happy to help**
HAPPY CODING :computer:

File: enumerable.rb [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+      my_each { |item| return true if item }
+    elsif args.instance_of?(Class)
+      my_each { |_item| return true if items.instance_of?(args) == args }
+    elsif args.instance_of?(Regexp)
+      my_each { |item| return true if args.match(item) }
+    else
+      my_each { |item| return true if item == args }
+    end
+    true
+  end
```
Comment (soieducation) [inline enumerable.rb]: - [ ]  This method returns true for all test cases, kindly fix this
Screenshot (305)

**Review by soieducation — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🚦 

## Hello @  :handshake: :handshake: :handshake: 
 Great work so far, you are almost there

### However, there are still changes to be made

# Required Changes :warning: 
- [ ] Check inline comments
___
**If you are stuck at any point or have any questions, let us know by leaving a comment here. We will be happy to help**
HAPPY CODING :computer:

**Review by soieducation — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @Ademola101   :wave: 

### You have done an amazing job on this project, congratulations! Your project is approved :clap: :clap: :clap: 

Good luck with your next project :rocket: 

### Remember to submit your project completion form
___
**If you are stuck at any point or have any questions, let us know by leaving a comment here. We will be happy to help**
HAPPY CODING :computer:

**Comment by Ademola101** [comment]
Thank you for the review and the suggestions

**Pull request merged**